File: backend/src/models/poin.js
import { DataTypes } from "sequelize";
import db from "../config/database.js";

const Poin = db.define(
  "Poin",
  {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    poin: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
     productId: {
      type: DataTypes.STRING, 
      allowNull: false,
    },
    promoProductId: {
      type: DataTypes.STRING,
      allowNull: true,
    },
  },
  {
    timestamps: false,
    tableName: "poin",
  }
);

export default Poin;

File: backend-go/models/poin.go
package models

type Poin struct {
	ID             uint   `gorm:"primaryKey;autoIncrement"`
	Poin           int    `gorm:"not null"`
	ProductID      string `gorm:"type:varchar(255);not null"` // Changed to string type
	PromoProductID string `gorm:"type:varchar(255)"`          // Nullable string

	//Diskon
	Discount *Discount `gorm:"foreignKey:PoinID"` // One-to-one
}

func (Poin) TableName() string {
	return "poin"
}
